source/slang/slang-parameter-binding.cpp

```diff
@@ -1071,15 +1088,24 @@ static void addExplicitParameterBindings_GLSL(
     LayoutSemanticInfo semanticInfo;
     semanticInfo.index = 0;
     semanticInfo.space = 0;
+    semanticInfo.inputAttachmentIndex = ParameterAndLayoutConstants::kInputAttachmentIndexUnused;
 
     if( (foundResInfo = typeLayout->FindResourceInfo(LayoutResourceKind::DescriptorTableSlot)) != nullptr )
     {
-        // Try to find `binding` and `set`
-        if (auto attr = varDecl.getDecl()->findModifier<GLSLBindingAttribute>())
+        for (auto dec : varDecl.getDecl()->modifiers)
         {
-            resInfo = foundResInfo;
-            semanticInfo.index = attr->binding;
-            semanticInfo.space = attr->set;
+            // Try to find `binding` and `set`
```

Comment:
I think inputattachment index should exist as a ResourceInfo entry rather than as an augmentation to DescriptorTableSlot. 

InputAttachment comes from framebuffer and not a descriptor table, IIRC.

Reply:
[input_attachment's are part of the renderpass, but they also need to be bound to a descriptor set and can have conflicts with other objects from what I see in LunarG's samples](https://github.com/LunarG/VulkanSamples/blob/master/API-Samples/input_attachment/input_attachment.cpp)

Reply:
Can you be more specific about the descriptor set? You can not expect us to understand what you meant by giving us a link to a source file. And please give us more explanation about the "conflict" you were talking about.

Reply:
The following comment was addressed in connection with other suggestions, a new ResourceInfo entry was added. A deeper explanation was provided to answer another comment, and will be provided (added to slightly) below:

_what each qualifier means:_

1. input_attachment_index references the index set when setting up renderpass's:
 ```
VkAttachmentReference renderPassAttachment{};
colorReference.attachment = 0;
```

2. binding & set is referencing the descriptor set and where we bound the renderpass's image view.
```
    VkDescriptorSetLayoutBinding layout_bindings[1];
    layout_bindings[0].binding = 0;
    layout_bindings[0].descriptorType = VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT;
    layout_bindings[0].descriptorCount = 1;
    layout_bindings[0].stageFlags = VK_SHADER_STAGE_FRAGMENT_BIT;
    layout_bindings[0].pImmutableSamplers = NULL;
```

_why we need them both for error reporting:_
1. if we bound to `descriptor set 1, binding 0, attachment 0` and `descriptor set 0, binding 0, attachment 0`, we should warn since this is a conflict which [spec warns is illegal for input_attachment_index](https://github.com/KhronosGroup/GLSL/blob/main/extensions/khr/GL_KHR_vulkan_glsl.txt).
```
layout (input_attachment_index = 0, set = 0, binding = 0) uniform subpassInput inputColor;
layout (input_attachment_index = 0, set = 1, binding = 0) uniform subpassInput inputColor2;
```

2. similarly, the same code here should conflict, but due to binding being the same, which is a problem since this is an impossible scenario to have (your descriptor is either for `input_attachment_index = 1` or `input_attachment_index = 0`, not both).
```
layout (input_attachment_index = 1, set = 0, binding = 0) uniform subpassInput inputColor;
layout (input_attachment_index = 0, set = 0, binding = 0) uniform subpassInput inputColor2;
```

3. finally, since this is a regular descriptor, the following code will also conflict: due to an impossible scenario of having two resources which are different using the same descriptor bindings.
```
layout (set = 0, binding = 0) uniform image1D inputTexture1D;
layout (input_attachment_index = 0, set = 0, binding = 0) uniform subpassInput inputColor;
```